programming-languages.md

```diff
@@ -1,51 +1,107 @@
 ---
-title: Server-side programming languages
+title: Programming languages
 layout: recommendation
-expires: 2017-09-01
+expires: 2017-10-01
 ---
 
-All programming languages have their own strengths and weaknesses.
-This means that we want to use a range of programming languages in
-order to build services and supporting tools to meet users needs.
-This document does not apply to choosing 'off the shelf' software
-(open source or not).
+We want to use a range of programming languages at GDS because we think
+using the right tool for the job will give us the best chance of building
+services that best meet users’ needs. This document does not apply to
+choosing 'off the shelf' software (open source or not).
 
-In order to make it easier for developers to move around the organisation,
-to get more value out of developing shared components, to better hire and
-develop people, and to be great at operating applications, we want to focus
-on a small number of choices of programming languages for core
-'backend development' tasks.
+We want to focus on using a small number of programming languages for
+core software development tasks.
 
-These are (alphabetically): Java, Python, Ruby. These are successfully being
-used by teams at the moment, and we are confident in how to host and operate
-applications written in them. New development should normally be carried out
-in one of them.
+This should make it easier for developers to:
 
-There will be sensible reasons to go outside this core group. For example:
+- move around the organisation
+- develop shared components
+- improve their personal development
+- master how they operate applications
 
-- when we're extending an existing codebase/ecosystem
-- when scripting in a particular environment
-- when we're experimenting during an alpha (with an expectation that it's replaced by something from the core set for beta)
+## Core languages
 
-The set of languages in the core group will change over time. This is likely
-to be through the adoption of a new language into the core group, and with
-an existing language being put on hold for new work.
+Our core languages are __Java, Python and Ruby__.
 
-## Languages in use not in the core group
+We've chosen these 3 languages because they are successfully used by
+teams at the moment, and we are confident in how to host and operate
+applications written in them. We'd advise you to carry out new
+development in one of them.
+
+## Languages in use in specific contexts
+
+### JavaScript
+
+This recommendation only applies to server-side JavaScript.
+The service manual has information on
+[using client-side JavaScript][manual_js].
+
+[manual_js]: https://www.gov.uk/service-manual/technology/using-progressive-enhancement
+
+The only way we're comfortable writing server-side JavaScript at the
+moment is with [Node.js][nodejs].
+
+[nodejs]: https://nodejs.org/
+
+Node.js is used by a few projects at GDS; GOV.UK Frontend, GOV.UK Pay and the Performance Platform.
```

Comment:
Can we make clear that "These projects have made the decision to use Node.JS, and the tech forum is watching the output of this.  No new projects should use Node.JS without prior consultation with Michael Brunton-Spall"